angelsexploration/control.lua

```diff
@@ -0,0 +1,46 @@
+local gathering_turret = require "src.gathering-turret"
+
+-- initialisation
+script.on_init(function()
+  gathering_turret:on_init()
+end)
+
+-- behaviour events
+script.on_event(defines.events.on_tick, function()
+  gathering_turret:on_tick_update()
+end)
+
+-- built events
+script.on_event(defines.events.on_built_entity, function(event)
+  gathering_turret:on_build_entity(event.created_entity)
+end)
+script.on_event(defines.events.on_robot_built_entity, function(event)
+  gathering_turret:on_build_entity(event.created_entity)
+end)
+script.on_event(defines.events.script_raised_built, function(event)
+  gathering_turret:on_build_entity(event.entity)
+end)
+script.on_event(defines.events.script_raised_revive, function(event)
+  gathering_turret:on_build_entity(event.entity)
+end)
+
+-- destroy events
+script.on_event(defines.events.on_entity_damaged, function(event)
+  gathering_turret:on_damaged_entity(event.entity, event.final_health)
+end, {
+  {filter = "name", name = "gathering-turret"},
+  {filter = "final-health", comparison = "=", value = 0},
+})
```

Comment:
Note to future self for gathering behavior rework:
```lua
script.on_event(defines.events.on_entity_damaged, function(event)
  gathering_turret:on_damaged_entity(event.entity, event.final_health)
end, {
  -- Event raise filter 1: damage to (inactive) gathering turret with 0 health remaining
  {mode = "or", filter = "name", name = "gathering-turret"},
  {mode = "and", filter = "final-health", comparison = "=", value = 0},
  -- Event raise filter 2: gathering damage to a potential gathering target
  {mode = "or", filter = "type", name = "land-mine"},
  {mode = "and", filter = "damage-type", name = "gathering"},
})
```

Reply:
Resolved in https://github.com/Arch666Angel/mods/pull/612/commits/46d6a4e1029c6efb7ceb82831c442ce8f4937c5a